app/views/urls/index.html.erb

```diff
@@ -0,0 +1,21 @@
+
+<h1>Welcome to our awesome url shortener</h1>
+<h2>Enter the url you want to shorten</h2>
+
+
+<%= form_for @url, url: {action: "create"}, html: {class: "entry_form"} do |f| %>
```

Comment:
I'd break this out into a partial, and you can use url.errors (which is a hash of symbols -> error objects) to show the errors.